ddtrace/internal/sampling.py

```diff
@@ -180,3 +189,67 @@ def apply_span_sampling_tags(self, span):
         span.set_metric(_SINGLE_SPAN_SAMPLING_RATE, self._sample_rate)
         if self._max_per_second:
             span.set_metric(_SINGLE_SPAN_SAMPLING_MAX_PER_SEC, self._max_per_second)
+
+
+class SpanSamplingRules:
+    """To generate a list of SpanSamplingRule objects and pass them to the SpanProcessor"""
+
+    def __init__(self):
+        self._rules = self._make_rules()
+
+    def get_single_span_sampling_rules(self):
+        # type: () -> List[SpanSamplingRule]
+        return self._rules
+
+    def _make_rules(self):
+        # type: () -> List[SpanSamplingRule]
+        json_rules = os.getenv("DD_SPAN_SAMPLING_RULES")
+        if not json_rules:
+            return []
+        else:
+            return self._parse_rules(json_rules)
+
+    def _parse_rules(self, rules):
+        sampling_rules = []
+        if rules is not None:
+            json_rules = []
+            try:
+                json_rules = json.loads(rules)
+            except JSONDecodeError:
+                raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES={}".format(rules))
+            for rule in json_rules:
+                sample_rate = rule.get("sample_rate", 1.0)
+                service = rule.get("service")
+                name = rule.get("name")
+                max_per_second = rule.get("max_per_second")
+                if service is None and name is None:
+                    raise SingleSpanSamplingError("Neither service or name specified for single span sampling rule")
```

Comment:
we should consider including `rule` in the message, so they know which one to look at